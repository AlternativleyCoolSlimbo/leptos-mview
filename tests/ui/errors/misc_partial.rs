use leptos_mview::mview;

fn invalid_value() {
    _ = mview! {
        div class:x={true} {
            span class=test
        }
    }
}

fn incomplete_directive() {
    _ = mview! {
        div class:x={true} {
            span class:
        }
    }
}

fn main() {}
