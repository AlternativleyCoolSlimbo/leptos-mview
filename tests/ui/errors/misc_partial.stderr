error: expected value after =
 --> tests/ui/errors/misc_partial.rs:6:24
  |
6 |             span class=test
  |                        ^^^^
  |
  = help: you may have meant to wrap this in braces

error: unterminated element
 --> tests/ui/errors/misc_partial.rs:6:13
  |
6 |             span class=test
  |             ^^^^
  |
  = help: add a `;` to terminate the element with no children

error: unexpected end of input, expected a kebab-cased ident
  --> tests/ui/errors/misc_partial.rs:15:9
   |
15 |         }
   |         ^

error[E0277]: the trait bound `MissingValueAfterEq: IntoAttribute` is not satisfied
 --> tests/ui/errors/misc_partial.rs:6:24
  |
4 |       _ = mview! {
  |  _________-
5 | |         div class:x={true} {
6 | |             span class=test
  | |                        ^^^^ the trait `Fn()` is not implemented for `MissingValueAfterEq`, which is required by `MissingValueAfterEq: IntoAttribute`
7 | |         }
8 | |     }
  | |_____- required by a bound introduced by this call
  |
  = help: the following other types implement trait `IntoAttribute`:
            &'static str
            &std::string::String
            Arguments<'_>
            Cow<'static, str>
            Nonce
            Oco<'static, str>
            Rc<str>
            TextProp
          and $N others
  = note: required for `MissingValueAfterEq` to implement `IntoAttribute`
note: required by a bound in `leptos::HtmlElement::<El>::attr`
 --> $CARGO/leptos_dom-0.6.12/src/html.rs
  |
  |     pub fn attr(
  |            ---- required by a bound in this associated function
...
  |         attr: impl IntoAttribute,
  |                    ^^^^^^^^^^^^^ required by this bound in `HtmlElement::<El>::attr`
